rcl_logging_spdlog/test/test_logging_interface.cpp

```diff
@@ -39,12 +95,71 @@ TEST(logging_interface, init_invalid)
   rcutils_reset_error();
 }
 
-TEST(logging_interface, full_cycle)
+TEST_F(LoggingTest, init_failure)
+{
+  RestoreEnvVar home_var("HOME");
+  RestoreEnvVar userprofile_var("USERPROFILE");
+
+  // No home directory to write log to
+  ASSERT_EQ(true, rcutils_set_env("HOME", nullptr));
+  ASSERT_EQ(true, rcutils_set_env("USERPROFILE", nullptr));
+  EXPECT_EQ(2, rcl_logging_external_initialize(nullptr, allocator));
```

Comment:
@cottsay 2? Don't we have a `#define` for it? Same elsewhere.

Reply:
Here is some context on that value: https://github.com/ros2/rcl_logging/pull/36#discussion_r425471405

I also discussed this offline (I'll ping you in the thread).

Reply:
Oh my, that's awful. Ok to move forward with this PR, but we need to address this for G-Turtle.

Reply:
Yeah, the story here is a bit of a mess. The plan is to do some package-level restructuring in G-turtle to de-duplicate the headers and definitions. I'd also like to provide a unified approach to testing the loggers.

Reply:
We've also been discussing it over at https://github.com/ros2/rcl_logging/pull/33#issuecomment-629416344 .  Definitely needs a cleanup.